tapdb/sqlc/migrations/000035_supply_tree.up.sql

```diff
@@ -0,0 +1,46 @@
+-- Ensure the required proof types for supply sub-trees exist.
+-- Note: 'issuance' is assumed to exist from previous migrations.
+INSERT INTO proof_types (proof_type) VALUES ('burn'), ('ignore'), ('mint_supply')
+    ON CONFLICT (proof_type) DO NOTHING;
```

Comment:
> We also add a new proof type of mint_supply, to distinguish from the normal issuance type.

Is the new proof type (`mint_supply`) necessary here? I thought we already added supply commit specific prefixes to the relevant namespaces? If true, then I think we could just reuse the `issuance` proof type.

Reply:
So I ended up using an entirely distinct namespace schema for this set of trees. See the `rootSupplyNamespace` and `subTreeNamespace` functions. This helped me to track down some bugs where because a distinct namespace wasn't used everywhere, sometimes a leaf would end up in the wrong tree, causing a test to fail. 

I added this extra proof type to distinguish the supply tree leaf, from the normal issuance leaf. Any public asset will have an issuance leaf, but only a subset of those will have a supply tree leaf.

Reply:
We still need to distinguish on the `proof_types` level as that table is referenced from elsewhere.